### PR TITLE
Improvement: Use SVGCanvas in Process page instead of SMCanvas

### PR DESCRIPTION
**Pull request by bbaa77770**
[LU-846](https://snapmaker2.atlassian.net/browse/LU-846)

### CODE REVIEW

File: src/app/ui/SVGEditor/SVGEditor.jsx [hunk 1]
```diff
@@ -19,6 +20,7 @@ class SVGEditor extends PureComponent {
         coordinateMode: PropTypes.object.isRequired,
         coordinateSize: PropTypes.object.isRequired,
         use3DVisualizer: PropTypes.bool,
+        page: PropTypes.string, // add cnc, add isRequired
```
Comment (meiliang1123) [inline src/app/ui/SVGEditor/SVGEditor.jsx:23]: Editor should not know page status, use props like `disableEdit` to switch editor editing state

**Review by meiliang1123 — CHANGES_REQUESTED** [review]

File: src/app/ui/SVGEditor/SVGEditor.jsx [hunk 2]
```diff
@@ -129,11 +130,13 @@ class SVGEditor extends PureComponent {
                             elementActions={this.props.elementActions}
                         />
                     </div>
-                    <SvgTool
-                        mode={this.state.mode}
-                        insertDefaultTextVector={this.insertDefaultTextVector}
-                        setMode={this.setMode}
-                    />
+                    {(!this.props.editDisabled) && (
```
Comment (parachvte) [inline src/app/ui/SVGEditor/SVGEditor.jsx:133]: remove brackets

File: src/app/ui/SVGEditor/svg-content/SVGContentGroup.js [hunk 3]
```diff
@@ -34,6 +35,7 @@ class SVGContentGroup {
             getRoot: () => this.svgContent,
             scale: this.scale
         });
+        this.operatorPoints.showGrips(true);
```
Comment (parachvte) [inline src/app/ui/SVGEditor/svg-content/SVGContentGroup.js:38]: ?

File: src/app/ui/SVGEditor/svg-content/SVGContentGroup.js [hunk 4]
```diff
@@ -493,8 +507,16 @@ class SVGContentGroup {
         return transformList[idx].angle;
     }
 
+    setEditDisabled(editDisabled) {
```
Comment (parachvte) [inline src/app/ui/SVGEditor/svg-content/SVGContentGroup.js:510]: Reconsider

File: src/app/ui/SVGEditor/svg-content/SVGContentGroup.js [hunk 5]
```diff
@@ -167,6 +169,9 @@ class SVGContentGroup {
         }
         const posAndsize = this.operatorPoints.resizeGrips(this.selectedElements);
         this.showSelectorGrips(true);
+        if (this.editDisabled) {
```
Comment (parachvte) [inline src/app/ui/SVGEditor/svg-content/SVGContentGroup.js:172]: Reconsider

Comment (parachvte) [reply src/app/ui/SVGEditor/svg-content/SVGContentGroup.js:172]: Call

```JS
this.operatorPoints.resizeGrips(this.selectedElements, !this.editDisabled)
```

definition:

```JS
resizeGrips(elements, showGrips = true) {
}
```

File: src/app/ui/SVGEditor/svg-content/SVGContentGroup.js [hunk 6]
```diff
@@ -13,10 +13,11 @@ class SVGContentGroup {
     selectedElements = [];
 
     constructor(options) {
-        const { svgContent, scale } = options;
+        const { svgContent, scale, editDisabled } = options;
 
         this.svgContent = svgContent;
         this.scale = scale;
+        this.editDisabled = editDisabled;
```
Comment (parachvte) [inline src/app/ui/SVGEditor/svg-content/SVGContentGroup.js:20]: editable

**Review by parachvte — CHANGES_REQUESTED** [review]
Changes required.

File: src/app/flux/editor/actions-process.js [hunk 7]
```diff
@@ -19,9 +19,11 @@ export const processActions = {
         });
     },
 
-    preview: (headType) => (dispatch) => {
+    preview: (headType) => (dispatch, getState) => {
         dispatch(processActions.recalculateAllToolPath(headType));
         dispatch(processActions.showToolPathGroupObject(headType));
+        const { SVGActions } = getState()[headType];
+        SVGActions.clearSelection();
```
Comment (parachvte) [inline src/app/flux/editor/actions-process.js:26]: Make a note. I think clear selection is wanted behavior.

Only when we can select multiple objects in edit and select single object only in process? @sumibin

File: src/app/ui/SVGEditor/SVGCanvas.jsx [hunk 8]
```diff
@@ -690,6 +696,9 @@ class SVGCanvas extends PureComponent {
             case 'move': {
                 const dx = x - draw.startX;
                 const dy = y - draw.startY;
+                if (this.props.editDisabled) {
```
Comment (parachvte) [inline src/app/ui/SVGEditor/SVGCanvas.jsx:699]:
```JS
if (this.props.editDisabled)
```

wrap the whole switch?

**Pull request merged**